pkg/download/jsoncreator/json_creator.go

```diff
@@ -131,6 +132,33 @@ func replaceKeyProperties(dat map[string]interface{}) map[string]interface{} {
 	return dat
 }
 
+// removes key with specified path
+func removeKey(dat map[string]interface{}, key []string) map[string]interface{} {
+	if len(key) == 0 {
```

Comment:
I still believe you could improve the readability of this a lot, if you remove some of the nesting / else block and just return early out of ifs. 

E.g. the method could look like this, turning all the initial checks into a (for me) much more readable sequence than the else blocks do:
```go

	if len(key) == 0 {
		// invalid input, nothing to remove
		return dat
	}

	if dat[key[0]] == nil {
		// no field of this name, nothing to do
		return dat
	}

	if len(key) == 1 {
		delete(dat, key[0])
		return dat
	}

	if field, ok := dat[key[0]].(map[string]interface{}); ok {
		dat[key[0]] = removeKey(field, key[1:])
	} else if arrayOfFields, ok := dat[key[0]].([]interface{}); ok {
		for i := range arrayOfFields {
			if field, ok := arrayOfFields[i].(map[string]interface{}); ok {
				arrayOfFields[i] = removeKey(field, key[1:])
			}
		}
		dat[key[0]] = arrayOfFields
	} else {
		util.Log.Error("Unable to parse configuration to remove identifier property %v. Correct configuration manually if necessary.", key[0])
	}

	return dat
```

(you could - and if I was writing this I would - drive this further and do the same with the last if/else-if/else block. But it doesn't help readability as much as the initial three IMHO)